chrome/content/addFinger.js

```diff
@@ -0,0 +1,41 @@
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("resource:///modules/imXPCOMUtils.jsm");
+Cu.import("chrome://otr/content/otr.js");
+
+XPCOMUtils.defineLazyGetter(this, "_", () =>
+  l10nHelper("chrome://otr/locale/addfinger.properties")
+);
+
+var args = window.arguments[0].wrappedJSObject;
+
+var otrAddFinger = {
+  onload: function() {
+    document.title = _("addfinger.title", args.screenname);
+  },
+  
+  oninput: function(e) {
+    e.value = e.value.replace(/[^0-9a-fA-F]/gi, "");
+    document.documentElement.getButton("accept").disabled = (e.value.length != 40);
+  },
+
+  add: function(e) {
+    let hex = document.getElementById("finger").value;
+    // remove exist fingerprint firstly
+    let finger = otr.hasContactFingerprint(args);
+    if (finger) {
+      finger.purge = true;
+      otr.forgetFingerprints([finger]);
+    }
+    let context = otr.getContextFromRecipient(
+          args.account,
+          args.protocol,
+          args.screenname
+        );
+    finger = otr.addFingerprint(context, hex);
+    if (!finger.isNull()) {
+      otr.setTrust(finger, true);
```

Comment:
Since you've added the ability to get to this window from the context menu, we also need to consider the case where you're already in an OTR session (we ignored that before since it was while adding a contact). You'll need to *try* to call `getUIConvFromContext()` and only if you get one, pass the context to `otr.setTrust`.

Also, please remove the `{}` around the single line if condition.